packages/saas-ui-auth/src/services/supabase.ts

```diff
@@ -6,103 +6,190 @@ import {
+    }
+
+    const resp = await supabase.auth.signUp({
+      email,
+      password,
+      phone,
+      options: {
+        captchaToken,
+        emailRedirectTo,
         data,
-        ...options,
-      }
-    )
+      },
+    });
 
-    if (user) {
-      return user as User
-    } else if (error) {
-      throw error
+    if (resp.error) {
+      throw resp.error;
     }
-  }
+    return resp.data.user;
+  };
 
   const onVerifyOtp = async (params: AuthParams) => {
-    const { session, error } = await supabase.auth.verifyOTP(params)
+    // redirectTo & captcha are part of the options, but not for AuthParams, maybe add there?
+    const { email, phone, otp, type, redirectTo, captchaToken } = params;
+    if (!otp) {
+      throw new Error(`Need otp to verify`);
+    }
 
-    if (session) {
-      return !!session
-    } else if (error) {
-      throw error
+    if (email) {
+      const verify: VerifyEmailOtpParams = {
+        email,
+        token: otp,
+        // This is not in the current api not sure what a good default would be
+        type: type || "signup",
+      };
+      const resp = await supabase.auth.verifyOtp(verify);
+      if (resp.error) {
+        throw resp.error;
+      }
+      return Boolean(resp.data.session);
+    }
+
+    if (phone) {
+      const verify: VerifyMobileOtpParams = {
+        phone,
+        token: otp,
+        // This is not in the current api not sure what a good default would be
+        type: type || "sms",
+        options: {
+          redirectTo,
+          captchaToken,
+        },
+      };
+      const resp = await supabase.auth.verifyOtp(verify);
+      if (resp.error) {
+        throw resp.error;
+      }
+      return Boolean(resp.data.session);
     }
-  }
+
+    throw new Error("You need to provide either email or phone");
+  };
 
   const onLogout = async () => {
-    return await supabase.auth.signOut()
-  }
+    return await supabase.auth.signOut();
+  };
 
   const onAuthStateChange = (callback: AuthStateChangeCallback) => {
     const { data } = supabase.auth.onAuthStateChange(
-      (event: any, session: any) => {
-        callback(session?.user as User)
+      (event: AuthChangeEvent, session: Session | null) => {
+        callback(session?.user);
       }
-    )
+    );
 
-    return () => data?.unsubscribe()
-  }
+    return () => data?.subscription.unsubscribe();
+  };
 
   const onLoadUser = async () => {
-    return await supabase.auth.user()
-  }
+    const { data, error } = await supabase.auth.getUser();
+    if (error) {
+      throw error;
+    }
+    return data.user;
+  };
 
   const onGetToken = async () => {
-    const session = supabase.auth.session()
-    return session?.access_token || null
-  }
+    const { data, error } = await supabase.auth.getSession();
+    if (error) {
+      throw error;
+    }
+    return data.session?.access_token || null;
+  };
 
   const onResetPassword = async ({ email }: AuthParams) => {
```

Comment:
Email is required, but it's optional in AuthParams, we should change the type here.

Reply:
Made a special params type for onResetPassword params.